gcp/appengine/Dockerfile

```diff
@@ -0,0 +1,51 @@
+# Build the Javascript frontend
+FROM node:latest AS FRONTEND3_BUILD
+WORKDIR /build/frontend3
+
+# Install dependencies first for better caching
+COPY gcp/appengine/frontend3/package.json gcp/appengine/frontend3/package-lock.json ./
+RUN npm ci
+
+COPY gcp/appengine/frontend3/webpack.prod.js ./
+COPY gcp/appengine/frontend3/img img
+COPY gcp/appengine/frontend3/src src
+
+RUN npm run build:prod
+
+# Build hugo blogs
+FROM klakegg/hugo:0.111.3 AS HUGO_BUILD
```

Comment:
Are there any alternatives here to relying on this unofficial image?

Since Hugo is a Go program, can we just pull the binary from an official release instead in some minimal image?

Reply:
or even build it from source if it's fast enough. that way we eliminate a few layers of potential compromise.

Reply:
I've changed it to build from source within the image, which takes some time but *should* be cached.
I could also just use the ci image, which builds Hugo (or make a bespoke Hugo image) if the caching doesn't work as intended.